src/products/components/ProductVariantChannels/ChannelsAvailabilityCard/AvailabilityCard.tsx

```diff
@@ -1,31 +1,45 @@
+import {
+  ChannelPriceAndPreorderData,
+  IChannelPriceAndPreorderArgs,
+} from "@dashboard/channels/utils";
 import { Divider } from "@dashboard/components/Divider";
+import { FormsetData } from "@dashboard/hooks/useFormset";
 import React from "react";
 
-import { Channel, ProductChannelListing } from "./../types";
+import { ProductChannelListing } from "./../types";
 import { ChannelsListItem } from "./ChannelsListItem";
 import CardContainer from "./VariantDetailsChannelsAvailabilityCardContainer";
 
 interface AvailabilityCardProps {
-  items: Channel[];
+  listings: FormsetData<
+    ChannelPriceAndPreorderData,
+    IChannelPriceAndPreorderArgs
+  >;
   productChannelListings: ProductChannelListing;
 }
 
 export const AvailabilityCard: React.FC<AvailabilityCardProps> = ({
-  items,
+  listings,
   productChannelListings,
   children,
 }) => {
-  if (items.length === 0) {
+  if (listings.length === 0) {
     return <CardContainer cardTitle={children}>{}</CardContainer>;
   }
 
+  const filteredListings: ProductChannelListing =
+    productChannelListings?.filter((channel: ProductChannelListing[0]) =>
+      listings.map(lst => lst.id).includes(channel.channel.id),
+    );
```

Comment:
suggestion: you do `map` for each  item in `productChannelListings`
Seem similar filtering is used in two places, would be better to move this filtering to function

```suggestion
const listingIds = listings.map(lst => lst.id);
 const filteredListings: ProductChannelListing =
    productChannelListings?.filter((channel: ProductChannelListing[0]) => listingIds.includes(channel.channel.id),
    );
 ```